manager/integration/tests/test_zone.py

```diff
@@ -0,0 +1,190 @@
+import pytest
+import time
+
+from common import client # NOQA
+from common import core_api  # NOQA
+from common import volume_name # NOQA
+from common import create_and_check_volume
+from common import get_self_host_id
+from common import wait_for_volume_replica_count
+from random import randrange
+from common import wait_for_volume_healthy
+from common import wait_for_volume_degraded
+from test_scheduling import wait_new_replica_ready
+from common import RETRY_COUNTS
+from common import RETRY_INTERVAL
+
+
+ZONE1 = "lh-zone1"
+ZONE2 = "lh-zone2"
+
+K8s_ZONE_LABEL = "topology.kubernetes.io/zone"
+
+
+@pytest.fixture
+def k8s_node_zone_tags(client, core_api):  # NOQA
+
+    lh_nodes = client.list_node()
+
+    node_index = 0
+    for node in lh_nodes:
+        node_name = node.name
+
+        if node_index % 2 == 0:
+            zone = ZONE1
+        else:
+            zone = ZONE2
+
+        payload = {
+            "metadata": {
+                "labels": {
+                    K8s_ZONE_LABEL: zone}
+            }
+        }
+
+        core_api.patch_node(node_name, body=payload)
+        node_index += 1
+
+    yield
```

Comment:
This fixture/generator is not invoked in the test. Will the following (cleanup) part be executed after the test complete?

Reply:
yes, `By using a yield statement instead of return, all the code after the yield statement serves as the teardown code:`
https://docs.pytest.org/en/latest/fixture.html#fixture-finalization-executing-teardown-code